docs/cql/time-to-live.rst

```diff
@@ -46,9 +46,20 @@ It’s also possible to set the TTL when performing an INSERT. To do this use:
 
         INSERT INTO heartrate(pet_chip_id, name, heart_rate) VALUES (c63e71f0-936e-11ea-bb37-0242ac130002, 'Rocky', 87) USING TTL 30;
 
-In this case, a TTL of 30 seconds is set. 
+In this case, a TTL of 30 seconds is set for each column. 
 
 
+When you set a TTL with an INSERT operation, the TTL will be applied to each individual column. If you UPDATE a column later without specifying a new TTL, the column's TTL will be updated to either 0 (no expiry) or the table-level TTL. If a column is updated without specifying a TTL, after the TTL that was set for the row during the INSERT operation has elapsed, all columns except the updated one will be expired. However, the row will still be available due to the presence of the unexpired column.
+
+.. code-block:: cql
+
+        INSERT INTO heartrate(pet_chip_id, name, heart_rate) VALUES (c63e71f0-936e-11ea-bb37-0242ac130002, 'Rocky', 87) USING TTL 30;
+
+        UPDATE heartrate SET heart_rate = 110 WHERE pet_chip_id = c63e71f0-936e-11ea-bb37-0242ac130002;
+
+After waiting for 30 seconds and running the SELECT command again, you will receive null values for all fields except for the heart_rate field. To expire the entire row after the initial TTL that was set during the INSERT operation, make sure to update the column with the appropriate TTL value.
+
```

Comment:
```suggestion

.. code-block:: cql

        UPDATE heartrate USING TTL 30 SET heart_rate = 110 WHERE pet_chip_id = c63e71f0-936e-11ea-bb37-0242ac130002;
```

Reply:
I'm not sure what's the point of this command (or the suggestion made above it). All the original columns of the row will have expired 30 seconds after they were originally written, but according to this command, the "heart_rate" set here will expire 30 seconds after **it** is written - not 30 seconds after the first write. So even with this command, you'll still have some time when most of the row has expired but the "heart_rate" column still hasn't.

The only guaranteed way to ensure that the entire row expires together in one fell swoop and mot different pieces of it at different times - is to rewrite the entire row again, including all its columns and not just the changed one. Use "INSERT" instead of "UPDATE" to also update the row-marker's TTL (this row marker thing is a subtle effect that's hard to explain, I can if you're curious, but isn't the main part of my suggestion).

Reply:
I've added the command to help the user understand the suggestion above it. Of course, now that you've explained why it is incorrect, the example should not be applied and the the suggestion should be rewritten.

Reply:
Hi @annastuchlik @nyh,
I have added a few examples on how to update the column with the appropriate TTL values.
Can help to take a look.

Reply:
@nyh @annastuchlik 
Can help to take a look. Thank you!